### ISSUE TITLE
Exception when trying to add a keybinding

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#23000]
- VSCode Version: Code - Insiders 1.11.0-insider (32942fd950cf2ffe19c44f944170014704b4fd2c, 2017-03-22T07:11:40.989Z)
- OS Version: Windows_NT ia32 10.0.15048
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|vscode-svgviewer|cssho|1.3.1|
|EditorConfig|EditorConfig|0.6.2|
|git-project-manager|felipecaputo|1.2.1|
|md-navigate|jrieken|0.0.1|
|vscode-scss|mrmlnc|0.6.2|
|seti-icons|qinjia|0.1.3|
|lorem-ipsum|Tyriar|1.0.0|
|terminal-tabs|Tyriar|0.1.1|
|theme-glacier|Tyriar|0.0.10|
|theme-sapphire|Tyriar|0.0.5|;

---

Steps to Reproduce:

![image](https://cloud.githubusercontent.com/assets/2193314/24189723/529c2948-0ea3-11e7-8e76-fae3baae52f4.png)

Got in this state, hit the pencil icon, pressed alt+enter, waited a few seconds and hit enter again to commit it. The keybindings tab is now completely frozen.

```
shell.ts:476Cannot read property 'toString' of null: TypeError: Cannot read property 'toString' of null
    at e.onResourceEncodingChange (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:123:7763)
    at file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:123:2539
    at e.invoke (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:31:7244)
    at e.fire (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:31:10877)
    at file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:118:7867
    at e.invoke (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:31:7244)
    at e.fire (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:31:10877)
    at t.loadWithContent (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:117:24196)
    at t.handleLoadSuccess (file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:117:23233)
    at file:///C:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/out/vs/workbench/electron-browser/workbench.main.js:117:23093e.onUnexpectedError @ shell.ts:476
```

**Comment by JimiC** [comment]
Related to https://github.com/Microsoft/vscode/issues/22996

**Comment by sandy081** [comment]
@Tyriar I am not able to reproduce this. Are there any particular steps I need to follow to get into this state?

**Comment by JimiC** [comment]
@sandy081 It has now been resolved via https://github.com/Microsoft/vscode/commit/301d64ef820359191c25198eba3dbb79a5e0c5d1
@Tyriar Can you confirm?

**Comment by Tyriar** [comment]
Still freezing on 6b4e8566

**Comment by sandy081** [comment]
Could be a dup of #23039

**Comment by sandy081** [comment]
@Tyriar I think this could be a dup of #23039. Please try it with the fix.

**Comment by Tyriar** [comment]
Still freezing on e1d2e7b

**Comment by sandy081** [comment]
@Tyriar Fix for #23039 went on Friday and not yet part of the insiders. It will be available in today's. Please check on today's build. Thanks.

**Comment by Tyriar** [comment]
It didn't freeze but I hit this:

```
messageService.ts:126Error 'Unable to write keybindings. Please open **Keybindings file** to correct errors/warnings in the file and try again.' while editing keybinding. Please open 'keybindings.json' file and check.
```

I had a missing comma in keybindings.json, it worked fine once I corrected it.
